viewing_party/party.py

```diff
@@ -1,23 +1,109 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie = {}
+    if title and genre and rating: 
+        movie["title"] = title
+        movie["genre"] = genre
+        movie["rating"] = rating
+        return movie
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if title in movie.values():
+            user_data["watchlist"].remove(movie)
+            user_data = add_to_watched(user_data, movie)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    sum = 0
+    movie_count = len(user_data["watched"])
+    if user_data["watched"]:
+        for movie in user_data["watched"]:
+            sum += movie["rating"]
+        return sum/movie_count
+    else:
+        return sum
 
+def get_most_watched_genre(user_data):
+    genres = []
+    if user_data["watched"]:
+        for movie in user_data["watched"]:
+            genres.append(movie["genre"])
+        # will return the genre with the highest count in the list
+        return max(genres, key = genres.count)
+    else:
+        return None
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    unique_movies = []
+    friend_count = len(user_data["friends"])
+    # Use friend_watched count to keep track of all the friends, to be sure that the movie
+    # is not in either friends watched list 
+    for movie in user_data["watched"]:
+        friend_watched = 0
+        for friend in user_data["friends"]:
+            if movie not in friend["watched"]:
+                friend_watched += 1
+        if friend_watched == friend_count:
+            unique_movies.append(movie)

```

Comment:
If we wanted to keep the variable named `friend_watched` the same, we cuold alter our if-statements to fit with the naming:

```suggestion
            if movie in friend["watched"]:
                friend_watched += 1
        if friend_watched == 0:
            unique_movies.append(movie)
```